packages/ui/src/utils/__tests__/TimeUtils.test.ts

```diff
@@ -0,0 +1,84 @@
+import { describe, expect, it } from 'vitest'
+
+import { formatDuration, formatTimestamp, getFormattedDuration } from '../TimeUtils'
+
+describe('getFormattedDuration', () => {
+  it('should return "0s" if startTs is greater than or equal to endTs', () => {
+    expect(getFormattedDuration(1000, 500)).toBe('0s')
+    expect(getFormattedDuration(1000, 1000)).toBe('0s')
+  })
+
+  it('should return formatted duration in "1h 2m 3s" format', () => {
+    expect(getFormattedDuration(0, 3723000)).toBe('1h 2m 3s')
+    expect(getFormattedDuration(0, 7200000)).toBe('2h')
+    expect(getFormattedDuration(0, 60000)).toBe('1m')
+    expect(getFormattedDuration(0, 1000)).toBe('1s')
+  })
+
+  it('should handle durations with only seconds', () => {
+    expect(getFormattedDuration(0, 5000)).toBe('5s')
+  })
+
+  it('should handle durations with hours, minutes, and seconds', () => {
+    expect(getFormattedDuration(0, 3661000)).toBe('1h 1m 1s')
+  })
+})
+
+describe('formatDuration', () => {
+  it('should return "0s" if duration is 0', () => {
+    expect(formatDuration(0)).toBe('0s')
+  })
+
+  it('should return formatted duration in "1h 2m 3s" format for milliseconds', () => {
+    expect(formatDuration(3723000, 'ms')).toBe('1h 2m 3s')
+  })
+
+  it('should return formatted duration in "1h 2m 3s" format for nanoseconds', () => {
+    expect(formatDuration(3723000000000, 'ns')).toBe('1h 2m 3s')
+  })
+
+  it('should handle durations with only milliseconds', () => {
+    expect(formatDuration(500, 'ms')).toBe('500ms')
+  })
+
+  it('should handle durations with only nanoseconds', () => {
+    expect(formatDuration(500000000, 'ns')).toBe('500ms')
+  })
+
+  it('should handle durations with hours, minutes, and seconds', () => {
+    expect(formatDuration(3661000, 'ms')).toBe('1h 1m 1s')
+  })
+})
+
+describe('formatTimestamp', () => {
```

Comment:
These tests don't actually _do_ anything, they're just asserting that the mock returns the hard-coded value. Instead of mocking `Intl.DateTimeFormat`, just set the timezone in `config/vitest-setup.ts` by adding `process.env.TZ = 'UTC'`. You can then change the expected values to be what they _should_ be in the last test (right now we're only passing the tests when it returns the wrong time)

Reply:
Yeah totally agreed. Fixed it as recommended above.